classes/observation_manager.php

```diff
@@ -0,0 +1,222 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * This file contains functions to get various observation data objects
+ *
+ * @package   mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace mod_observation;
+
+use coding_exception;
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * mod_observation observation management class
+ *
+ * @package   mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class observation_manager {
+    /**
+     * Modifies or creates a new observation point in the database
+     * @param mixed $data Data to pass to database function
+     * @param bool $newinstance True if new instance, else false if editing
+     * @param bool $returnid True if should return id (only when $newinstance = true)
+     * @param string $tablename Tablename
+     * @return mixed True if successful. If $returnid is True and $newinstance is True, returns ID
+     */
+    public static function modify_observation_point($data, bool $newinstance = false, bool $returnid = false,
+        string $tablename = 'observation_points') {
+
+        $data = (object)$data;
+
+        global $DB;
```

Comment:
Can we have the global declarations as the very top line in the functions? It can serve as a nice little flag as to the reach of a function